app/src/main/java/pl/szczodrzynski/edziennik/ui/dialogs/timetable/GenerateBlockTimetableDialog.kt

```diff
@@ -373,25 +376,31 @@ class GenerateBlockTimetableDialog(
 
             val today = Date.getToday().stringY_m_d
             val now = Time.getNow().stringH_M_S
-
-            val outputDir = Environment.getExternalStoragePublicDirectory("Szkolny.eu").apply { mkdirs() }
-            val outputFile = File(outputDir, "plan_lekcji_${app.profile.name}_${today}_${now}.png")
+            val filename = "plan_lekcji_${app.profile.name}_${today}_${now}.png"
+            val resolver: ContentResolver = activity.applicationContext.contentResolver
+            val values = ContentValues()
+            values.put(MediaStore.MediaColumns.MIME_TYPE, "image/png")
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                values.put(MediaStore.MediaColumns.DISPLAY_NAME, filename)
+                values.put(MediaStore.MediaColumns.RELATIVE_PATH, File(Environment.DIRECTORY_PICTURES, "Szkolny.eu").path)
+            } else {
+                val picturesDirectory = File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES).absolutePath, "Szkolny.eu")
```

Comment:
This is still not what I meant.
Take a look at the example code here:
https://developer.android.com/reference/android/os/Environment#getExternalStoragePublicDirectory(java.lang.String)
You don't have to get the absolutePath there. 
Also this method seems to be deprecated for some reason, so you may consider using one of the replacements mentioned in the docs.